crates/aranya-client/src/aqc/net.rs

```diff
@@ -0,0 +1,824 @@
+    /// Create an Aqc client with the given certificate chain.
+    pub fn new<T: provider::tls::Provider>(
+        provider: T,
+        client_keys: Arc<ClientPresharedKeys>,
+        identity_rx: mpsc::Receiver<Vec<u8>>,
+        server: Server,
+        daemon: Arc<DaemonApiClient>,
+    ) -> Result<AqcClient> {
+        let quic_client = Client::builder()
+            .with_tls(provider)?
+            .with_io("0.0.0.0:0")?
+            .start()?;
+        Ok(AqcClient {
+            quic_client,
+            client_keys,
+            channels: HashMap::new(),
+            server,
+            daemon,
+            identity_rx,
+        })
+    }
+
+    /// Get the local address of the client.
+    pub async fn local_addr(&self) -> Result<SocketAddr> {
+        Ok(self.quic_client.local_addr()?)
+    }
+
+    /// Receive the next available channel. If the channel is closed, return None.
+    /// This method will return a channel created by a peer that hasn't been received yet.
+    pub async fn receive_channel(&mut self) -> Result<AqcChannelType, AqcError> {
+        loop {
+            // Accept a new connection
+            match self.server.accept().await {
+                Some(mut conn) => {
+                    // Receive a PSK identity hint.
+                    // TODO: Instead of receiving the PSK identity hint here, we should
+                    // pull it directly from the connection.
+                    if let Some(identity) = self.identity_rx.recv().await {
+                        tracing::debug!(
+                            "Processing connection accepted after seeing PSK identity hint: {:02x?}",
+                            identity
+                        );
+                        // If the PSK identity hint is the control PSK, receive a control message.
+                        // This will update the channel map with the PSK and associate it with an
+                        // AqcChannel.
+                        if identity == PSK_IDENTITY_CTRL {
+                            receive_ctrl_message(&self.daemon, &mut self.channels, &mut conn)
+                                .await?;
+                        // If the PSK identity hint is not the control PSK, check if it's in the channel map.
+                        // If it is, create a channel of the appropriate type. We should have already received
+                        // the control message for this PSK, if we don't we can't create a channel.
+                        } else if let Some(channel_info) = self.channels.get(&identity) {
+                            return Ok(create_channel_type(conn, channel_info));
+                        } else {
+                            tracing::debug!(
+                                "No channel info found in map for identity hint {:02x?}",
+                                identity
+                            );
+                            return Err(AqcError::NoChannelInfoFound);
+                        }
```

Comment:
```suggestion
                        } else {
                            match self.channels.get(&identity) {
                                Some(channel_info) => {
                                    return Ok(create_channel_type(conn, channel_info))
                                }
                                None => {
                                    tracing::debug!(
                                        "No channel info found in map for identity hint {:02x?}",
                                        identity
                                    );
                                    return Err(AqcError::NoChannelInfoFound);
                                }
                            }
                        }
```
This should read better, wish I could combine it with the outer if/else but idk if there's a good way to without just nesting